internal/content_coding.go

```diff
@@ -264,6 +283,41 @@ func (e *ZlibEncoder) Encode(data []byte) ([]byte, error) {
 	return e.buf.Bytes(), nil
 }
 
+type ZstdEncoder struct {
+	encoder *zstd.Encoder
+}
+
+func NewZstdEncoder(options ...EncodingOption) (*ZstdEncoder, error) {
+	cfg := encoderConfig{level: -1}
+	for _, o := range options {
+		o(&cfg)
+	}
+
+	// Map the levels
+	var level zstd.EncoderLevel
+	switch cfg.level {
+	case 1:
+		level = zstd.SpeedFastest
+	case -1, 3:
+		level = zstd.SpeedDefault
+	case 6:
+		level = zstd.SpeedBetterCompression
+	case 9:
+		level = zlib.BestSpeed
+	default:
+		return nil, fmt.Errorf("invalid compression level, only 1, 3, 6 and 9 are supported")
```

Comment:
Where are these numbers coming from? The library mentions these only.
![image](https://github.com/influxdata/telegraf/assets/87654537/7f3b1728-f08d-47b1-9321-a1dfd2ae9ac4)

Reply:
Maybe we can use this [function](https://github.com/klauspost/compress/blob/master/zstd/encoder_options.go#L188) and not bother that much about it.

Reply:
The linked function is exactly where the values are coming from except for the `9` which is "best compression" in the standard-levels, isn't it?

Reply:
Well the user will likely not find these... Anyway, if you feel better to go with those values, please adapt the code accordingly...

Reply:
And please remember to document the levels in a README....

Reply:
but Is it worth it to make a README file inside internal directory just to document the levels?

Reply:
Yes as we then can include this in all users (i.e. all plugins that expose the compression to the user) of the encoding/decoding stuff.